scarb/src/ops/fmt.rs

```diff
@@ -0,0 +1,182 @@
+use anyhow::Result;
+use cairo_lang_formatter::{CairoFormatter, FormatOutcome, FormatterConfig};
+use diffy::create_patch;
+use ignore::WalkState::Continue;
+use ignore::{DirEntry, Error, ParallelVisitor, ParallelVisitorBuilder, WalkState};
+use indicatif::HumanDuration;
+use std::path::Path;
+use std::sync::Mutex;
+use tracing::info;
+
+use crate::core::workspace::Workspace;
+use crate::core::{Package, PackageName, SourceKind};
+use crate::ops;
+use crate::ui::Status;
+
+#[derive(Debug)]
+pub struct FmtOptions {
+    pub check: bool,
+    pub all: bool,
+    pub pkg_name: Option<PackageName>,
+    pub color: bool,
+}
+
+fn print_error(error: anyhow::Error, path: String, ws: &Workspace<'_>) {
+    ws.config().ui().error(format!(
+        "A parsing error occurred in {path}. The content was not formatted.\n{error}"
+    ));
```

Comment:
<img width="1151" alt="image" src="https://user-images.githubusercontent.com/3450050/217198033-13d26fb0-da79-4706-918c-841fbfeb5606.png">
Could we make this look like this?